File: SecondYear/Anantj2499_AnantJaiswal_2226cseml1088_2/Week_4/FirstMissingPositive.java
import java.util.* ;
import java.io.*;
public class FirstMissingPositive {
    public static int firstMissing(int[] arr, int n) {
        // Write your code here.

        Arrays.sort(arr);
        int count = 1;
        for (int i = 0; i < n; i++) {
            if(count>arr[i]) {
                continue;
            }
            else if(count==arr[i])
                count++;
            else
                break;
        }
        return count;
    }
}

File: SecondYear/sauban123_MohdSauban_2226csit1130_2/Week_4/FirstMissingPositive.cpp
#include <bits/stdc++.h> 
int firstMissing(int arr[], int n)
{
    // Write your code here
    vector<int> ans;
    for(int i=0 ;i<n ;i++){
        if(arr[i]>0){
            ans.push_back(arr[i]);
        }
    }
    sort(ans.begin(),ans.end());
    int m=ans.size();
    for(int i=0;i<m;i++){
        if(ans[i] !=i+1){
            return i+1;
        }
    }
    return m+1;
}
